### PR TITLE
Added relative duration metrics

### PR DESCRIPTION
**Pull request by eduard-sukharev**
Relative duration metrics are taken as a fraction of total samples time (not total benchmark time, but a total of all average for each benchmark sample), a fraction of slowest benchmark sample (essentially, showing how much faster all the other samples are, in % to the slowest one), and a fraction of fastest benchmark sample (showing how much slower, in % to the fastest one, all other samples are).

@tsantos84 waiting for your thought on whether some of these metrics are at least meaningful or anyhow useful. My thoughts are that they might provide more reliable reference metrics — percentage of the total shouldn't change significantly on different machines.

image

### CODE REVIEW

**Comment by tsantos84** [comment]
@eduard-sukharev nice addition to benchmark. Now we can start putting some inteligence in the application and get more insights when choose such library. What do you think instead of show `100`, `142` and `531` we display `0`, `42` and `808` respectively (only subtract 100 from each metric). This way, is very clear to me that simple-serializer vendor is 42% faster than symfony. The same comparisson would be applied in the last column, but within an inversed math. Something that tells me that jms is x% slowest than symfony. This math makes sense to you?

**Comment by eduard-sukharev** [comment]
That totally makes sense and seem to be what I initially wanted to get, but didn't know how 😋
So, whcih relative metrics should be left? Only the one proposed by you? Because the last one is redundant and kinda inverse and the first one isn't very obvious.

**Comment by eduard-sukharev** [comment]
@tsantos84 Ok, now it seem to be better:
image
It's clear that symfony serializer is only 50% slower, while JMS is 4-5 times slower than clear deserialization leader - Opensoft's SimpleSerializer!
image
For serialization it's clear that the winner is TSantos serializer. SimpleSerializer is about 1-1.5 times slower, while symfony and JMS being tiers at around 2 times slower (although, symfony seem to be a bit faster).

**Comment by tsantos84** [comment]
@eduard-sukharev seems good to me. Could you please check the travis report? Symfony's deserialization is outputting some warnings. BTW, shouldn't the (de)serializations assertions throws exceptions instead of warnings? Fix only the warnings for now and we can discuss this in another PR

File: src/Console/Command/SerializeCommand.php [hunk 1]
```diff
@@ -66,6 +66,6 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         $result = $this->benchmark->run($samples, $batchCount);
 
-        $style->table(['vendor', 'duration (ms)', 'duration, % of total', 'duration, % of fastest', 'duration, % of slowest'], $this->getHelper('result')->sort($result));
+        $style->table(['vendor', 'duration (ms)', 'relative to fastest, %'], $this->getHelper('result')->sort($result));
```
Comment (tsantos84) [inline src/Console/Command/SerializeCommand.php]: I think the column description should be `relative to fastest (%)`.

**Review by tsantos84 — COMMENTED** [review]
@eduard-sukharev for some rease, the serialization report is now working properly. Could you please see what is the problem? I've merged `master` into `improve-report` branch and fixed two simple conflicts.

**Comment by tsantos84** [comment]
My mistake. Merging to master

**Comment by eduard-sukharev** [comment]
sorry for delay - I've checked the branch and it worked for me, but I forgot to reply back :(

**Pull request merged**